geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/AvroIngestProcessor.scala

```diff
@@ -11,130 +11,86 @@ package org.geomesa.nifi.datastore.processor
-      val outLocalName = outAttributeDescriptor.getLocalName
-      in.indexOf(outLocalName) match {
-        case inIndex if inIndex >= 0 =>
-          val outTypeBinding = outAttributeDescriptor.getType.getBinding
-          if (outTypeBinding.isAssignableFrom(in.getType(inIndex).getBinding)) {
-            sf: SimpleFeature => sf.getAttribute(inIndex)
-          } else {
-            sf: SimpleFeature => Converters.convert(sf.getAttribute(inIndex), outTypeBinding).asInstanceOf[AnyRef]
-          }
-        case _ if outLocalName.equals(outGeometryLocalName) => sf: SimpleFeature => sf.getDefaultGeometry
-        case _ => _: SimpleFeature => null
-      }
-    }
-    sf: SimpleFeature => {
-      val o = SimpleFeatureBuilder.build(out, inToOut.map(_(sf)).asJava, sf.getID)
-      o.getUserData.putAll(sf.getUserData)
-      o
-    }
-  }
-}
-
 /**
   * Avro ingest processor for geotools data stores
   */
 @CapabilityDescription("Ingest GeoAvro files into GeoMesa")
-trait AvroIngestProcessor extends FeatureTypeProcessor {
+@ReadsAttributes(
+  Array(new ReadsAttribute(attribute = "geomesa.sft.name", description = "GeoMesa SimpleFeatureType name"))
+)
+trait AvroIngestProcessor extends DataStoreIngestProcessor {
+
+  import AvroIngestProcessor.Properties.UseProvidedFid
+  import AvroIngestProcessor.{ModeMappings, Properties}
+  import org.geomesa.nifi.datastore.processor.mixins.FeatureTypeProcessor.Properties.FeatureNameOverride
 
-  import AvroIngestProcessor.{AvroMatchMode, UseProvidedFid}
+  override protected def getPrimaryProperties: Seq[PropertyDescriptor] =
+    super.getPrimaryProperties ++ Seq(FeatureNameOverride, UseProvidedFid)
 
-  override protected def getProcessorProperties: Seq[PropertyDescriptor] =
-    super.getProcessorProperties ++ Seq(AvroMatchMode, UseProvidedFid)
+  // noinspection ScalaDeprecation
+  override protected def getConfigProperties: Seq[PropertyDescriptor] =
+    super.getConfigProperties ++ Seq(Properties.AvroMatchMode, Properties.AvroSftName)
 
+  // noinspection ScalaDeprecation
   override protected def createIngest(
       context: ProcessContext,
       dataStore: DataStore,
       writers: FeatureWriters,
-      sftArg: Option[String],
-      typeName: Option[String]): IngestProcessor = {
-    val matchMode = context.getProperty(AvroMatchMode).getValue
+      mode: CompatibilityMode): IngestProcessor = {
     val useProvidedFid = context.getProperty(UseProvidedFid).getValue.toBoolean
-    new AvroIngest(dataStore, writers, sftArg, typeName, matchMode, useProvidedFid)
+    val matchMode =
+      Option(context.getProperty(Properties.AvroMatchMode).getValue)
+          .collect { case m if m.nonEmpty && !ModeMappings.contains(m -> mode) =>
+            val compat = ModeMappings.collectFirst { case (old, c) if old == m => c }.get
+            logger.warn(s"Using deprecated match mode to override compatibility mode $mode with $compat")
+            compat
+          }
```

Comment:
Can this be None?

Reply:
i don't think so b/c it's coming from the 'allowableValues' drop down